crates/biome_js_analyze/src/lint/nursery/no_vue_duplicate_keys.rs

```diff
@@ -0,0 +1,193 @@
+    ///         }
+    ///     },
+    ///     methods: {
+    ///         clearMessage() {
+    ///             this.message = '';
+    ///         }
+    ///     }
+    /// };
+    /// </script>
+    /// ```
+    ///
+    pub NoVueDuplicateKeys {
+        version: "next",
+        name: "noVueDuplicateKeys",
+        language: "js",
+        recommended: true,
+        severity: Severity::Error,
+        domains: &[RuleDomain::Vue],
+        sources: &[RuleSource::EslintVueJs("no-dupe-keys").same()],
+    }
+}
+
+impl Rule for NoVueDuplicateKeys {
+    type Query = VueComponentQuery;
+    type State = RuleState;
+    type Signals = Box<[Self::State]>;
+    type Options = NoVueDuplicateKeysOptions;
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let Some(component) = VueComponent::from_potential_component(
+            ctx.query(),
+            ctx.model(),
+            ctx.source_type(),
+            ctx.file_path(),
+        ) else {
+            return Box::new([]);
+        };
+
+        let mut key_declarations: FxHashMap<String, Vec<VueDeclaration>> = FxHashMap::default();
+
+        // Collect all declarations across all Vue component sections
+        for declaration in component.declarations(VueDeclarationCollectionFilter::all()) {
+            if let Some(name) = declaration.declaration_name() {
+                let key = name.text().to_string();
+                key_declarations.entry(key).or_default().push(declaration);
+            }
+        }
+
+        // Find duplicates
+        key_declarations
+            .into_iter()
+            .filter_map(|(key, declarations)| {
+                if declarations.len() > 1 {
+                    Some(RuleState { key, declarations })
+                } else {
+                    None
+                }
+            })
+            .collect::<Box<[_]>>()
+    }
+
+    fn diagnostic(_ctx: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        let first_declaration = state.declarations.first()?;
+        let mut diagnostic = RuleDiagnostic::new(
+            rule_category!(),
+            first_declaration.declaration_name_range()?,
+            markup! {
+                "Duplicate key "<Emphasis>{&state.key}</Emphasis>" found in Vue component."
+            },
+        );
+
+        // Add related information for other occurrences
+        for declaration in state.declarations.iter().skip(1) {
+            if let Some(range) = declaration.declaration_name_range() {
+                diagnostic = diagnostic.detail(
+                    range,
+                    markup! {
+                        "Key "<Emphasis>{&state.key}</Emphasis>" is also defined here."
+                    },
+                );
+            }
+        }
+
+        diagnostic = diagnostic.note(markup! {
+            "Keys defined in different Vue component options (props, data, methods, computed) can conflict when accessed in the template."
+        });
```

Comment:
Don't forget to tell the user what to do to resolve the issue (even when it's obvious)

Reply:
Added, please check.